project/SalatBuild.scala

```diff
@@ -77,18 +77,19 @@ object BuildSettings {
     parallelExecution in Test := false,
     testFrameworks += TestFrameworks.Specs2,
     resolvers ++= Seq(typeSafeRepo, typeSafeSnapsRepo, oss, ossSnaps),
-    javacOptions ++= Seq("-source", "1.6", "-target", "1.6"), 
+    javacOptions ++= Seq("-source", "1.6", "-target", "1.6"),
     scalacOptions ++= Seq("-deprecation", "-unchecked", "-feature", "-language:_"),
     crossScalaVersions ++= Seq("2.10.6")
   )
 }
 
 object Scalariform {
 
-import com.typesafe.sbt.SbtScalariform._
-import scalariform.formatter.preferences._
+  import scalariform.formatter.preferences._
+  import com.typesafe.sbt.SbtScalariform
+  import com.typesafe.sbt.SbtScalariform.ScalariformKeys
 
-  val settings = scalariformSettings ++ Seq(
+  val settings = SbtScalariform.defaultScalariformSettings ++ Seq(
```

Comment:
I guess i wasn't detailed enough in my requirements. I want to upgrade sbt (and therefore scalariform), and I *do* want to format code on compilation - but I want it to maintain the existing formatting rules as closely as possible, so few files get reformatted.

Reformatting the entire project upon sbt upgrade is not my goal, and would put too much noise in the revision history.

I have this same problem at work with closed-source code at Novus, so I think that I will open an issue with scalariform.